src/Snapshot.php

```diff
@@ -90,45 +96,69 @@ protected function shouldIgnoreLine(string $line): bool
-            while (! feof($stream)) {
-                $chunk = $this->compressionExtension === 'gz'
+        $directory = (new TemporaryDirectory(config('db-snapshots.temporary_directory_path')))->create();
+
+        config([
+            'filesystems.disks.' . self::class => [
+                'driver' => 'local',
+                'root' => $directory->path(),
+                'throw' => false,
+            ]
+        ]);
+
+        $localDisk = $this->filesystemFactory->disk(self::class);
+
+        try {
+            LazyCollection::make(function () use ($localDisk) {
+                $localDisk->writeStream($this->fileName, $this->disk->readStream($this->fileName));
+
+                $stream = $this->compressionExtension === 'gz'
+                    ? gzopen($localDisk->path($this->fileName), 'r')
+                    : $localDisk->readStream($this->fileName);
+
+                $statement = '';
+                while (!feof($stream)) {
+                    $chunk = $this->compressionExtension === 'gz'
                         ? gzread($stream, self::STREAM_BUFFER_SIZE)
                         : fread($stream, self::STREAM_BUFFER_SIZE);
 
-                $lines = explode("\n", $chunk);
-                foreach ($lines as $idx => $line) {
-                    if ($this->shouldIgnoreLine($line)) {
-                        continue;
+                    $lines = explode("\n", $chunk);
+                    foreach ($lines as $idx => $line) {
+                        if ($this->shouldIgnoreLine($line)) {
+                            continue;
+                        }
+
+                        $statement .= $line;
+
+                        // Carry-over the last line to the next chunk since it
+                        // is possible that this chunk finished mid-line right on
+                        // a semi-colon.
+                        if (count($lines) == $idx + 1) {
+                            break;
+                        }
+
+                        if (str_ends_with(trim($statement), ';')) {
+                            yield $statement;
+                            $statement = '';
+                        }
                     }
+                }
 
-                    $statement .= $line;
-
-                    // Carry-over the last line to the next chunk since it
-                    // is possible that this chunk finished mid-line right on
-                    // a semi-colon.
-                    if (count($lines) == $idx + 1) {
-                        break;
-                    }
+                if (str_ends_with(trim($statement), ';')) {
+                    yield $statement;
+                }
 
-                    if (str_ends_with(trim($statement), ';')) {
-                        yield $statement;
-                        $statement = '';
-                    }
+                if ($this->compressionExtension === 'gz') {
+                    gzclose($stream);
+                } else {
```

Comment:
I have a profound hate towards using `else` 🙂  Could you either use a ternary or extract this to a function and use an early return to avoid the `else`?

Reply:
done